### PR TITLE
Fixed android permissions callback

### PR DESCRIPTION
**Pull request by l0stpenguin**

### CODE REVIEW

**Comment by ombr** [comment]
Could you provide more information on this change ?

**Comment by l0stpenguin** [comment]
> Could you provide more information on this change ?

The original plugin did not wait for the request permission callback as denoted in those comments:
```
//This method executes async and we seem to have no known way to receive the result
// that's why these methods were later added to Cordova), so simply returning ok now.
```
The methods were later added by cordova (which was not available when the orignal plugin was developed):
`cordova.requestPermissions`

File: src/android/com/synconset/ImagePicker/ImagePicker.java [hunk 1]
```diff
@@ -111,14 +111,10 @@ private boolean hasReadPermission() {
...
-                this.cordova.getActivity(),
-                new String[] {Manifest.permission.READ_EXTERNAL_STORAGE},
-                PERMISSION_REQUEST_CODE);
+            cordova.requestPermissions(this, PERMISSION_REQUEST_CODE, new String[] {Manifest.permission.READ_EXTERNAL_STORAGE});
+            return;
         }
-        // This method executes async and we seem to have no known way to receive the result
-        // (that's why these methods were later added to Cordova), so simply returning ok now.
-        callbackContext.success();
+        callbackContext.success(1);
```
Comment (azharbeebeejaun) [inline src/android/com/synconset/ImagePicker/ImagePicker.java:117]: tu aurais pu inverser la logique
if (hasReadPerm) {
...
} else {
...
}
pas besoin du return non plus

File: src/android/com/synconset/ImagePicker/ImagePicker.java [hunk 2]
```diff
@@ -155,19 +151,16 @@ public void onRestoreStateForActivityResult(Bundle state, CallbackContext callba
...
     @Override
-    public void onRequestPermissionResult(int requestCode,
-                                          String[] permissions,
-                                          int[] grantResults) throws JSONException {
+    public void onRequestPermissionResult(int requestCode, String[] permissions, int[] grantResults) throws JSONException {
 
         // For now we just have one permission, so things can be kept simple...
         if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-            cordova.startActivityForResult(this, imagePickerIntent, 0);
+            callbackContext.success(1);
```
Comment (azharbeebeejaun) [inline src/android/com/synconset/ImagePicker/ImagePicker.java:160]: on envoie 1 en cas de succès ? dans le else aussi c'est un cas de succès ?

Comment (l0stpenguin) [reply src/android/com/synconset/ImagePicker/ImagePicker.java:160]: oui la method est successful, le 1/0 represente la statue de permission.

**Pull request merged**